apidaze/applications/__init__.py

```diff
@@ -0,0 +1,107 @@
+from apidaze.http import HttpMethodEnum
+
+
+class Applications(object):
+    """
+        Initializes the Applications class
+
+        Parameters
+        ----------
+        http: Http
+            Apidaze Http object
+
+        Returns
+        -------
+        object
+            The Apidaze Applications object
+    """
+    def __init__(self, http):
+        self.http = http
+        self.endpoint = '/applications'
+
+    def list(self):
+        """
+            Shows sub-applications list
+
+            Returns
+            -------
+            dict
+                JSON response
+        """
+        return self.__prepare_request()
+
+    def get_by_app_id(self, app_id: int):
+        """
+            Shows application with id
+
+            Parameters
+            ----------
+            app_id: int
+                Id of the sub-application
+
+            Returns
+            -------
+            dict
+                JSON response
+        """
+        params = {
+            'app_id': app_id
+        }
+
+        return self.__prepare_request(params=params)
+
+    def get_by_api_key(self, api_key: str):
+        """
+            Shows application with api_key
+
+            Parameters
+            ----------
+            api_key: str
+                Api key of the sub-application
+
+
+            Returns
+            -------
+            dict
+                JSON response
+        """
+        params = {
+            'api_key': api_key
+        }
+
+        return self.__prepare_request(params=params)
+
+    def get_by_name(self, name: str):
+        """
+            Shows application with name
+
+            Parameters
+            ----------
+            name: str
+                Name of the sub-application
+
+            Returns
+            -------
+            dict
+                JSON response
+        """
+        params = {
+            'app_name': name
+        }
+
+        return self.__prepare_request(params=params)
+
+    def __prepare_request(self, params: dict = {}):
+        response = self.http.request(
+            method=HttpMethodEnum.GET,
+            endpoint=self.endpoint,
+            payload={},
+            params=params
+            )
+
+        result = {
+            'body': response.json(),
+            'status_code': response.status_code
+        }
```

Comment:
As far as I understood from the tickets, we should have helper methods which return an instance of `Client` with the fetched application by using (`get_by_app_id`, `get_by_api_key`, `get_by_name`).

For example for `get_by_name`, we may have a method as follows;

```python
# rest of the imports
from apidaze import Client


class Applications(object):
    # rest of the class

    def get_client_by_name(self, name: str):
        app_metadata = self.get_by_name(name=name)
        return Client(api_key=app_metada.api_key, api_secret=app_metada.api_secret)
```

Then one may consume a fetched application as below;

```python
root_client = Client(api_key="a_key", api_secret="a_secret")
root_client.get_client_by_name(name="my_script")
```